libraries/aws/shadow/src/aws_iot_shadow_api.c

```diff
@@ -218,52 +215,64 @@ static AwsIotShadowError_t _validateThingNameFlags( _shadowOperationType_t type,
         IotLogError( "Thing Name for Shadow %s is not valid.",
                      _pAwsIotShadowOperationNames[ type ] );
 
-        IOT_SET_AND_GOTO_CLEANUP( AWS_IOT_SHADOW_BAD_PARAMETER );
+        status = AWS_IOT_SHADOW_BAD_PARAMETER;
     }
 
     /* Check the waitable operation flag. */
     if( ( flags & AWS_IOT_SHADOW_FLAG_WAITABLE ) == AWS_IOT_SHADOW_FLAG_WAITABLE )
     {
-        /* Check that a reference pointer is provided for a waitable operation. */
-        if( pOperation == NULL )
+        if( status == AWS_IOT_SHADOW_SUCCESS )
```

Comment:
Is it possible to refactor this without having to have the success status checks through this functions? As we have discussed earlier, having a single stream of happy-path code is easier to read

Reply:
ok, let me try to refactor _validateThingNameFlags

Reply:
It doesn't look like any of the status checks in this function are required. All the errors change the status to `AWS_IOT_BAD_PARAMETER`, so if multiple paths are taken, the end result will be the same, and the only difference will be multiple error logs.

Reply:
I will refactor.